src/Symfony/Component/Ldap/Adapter/ExtLdap/Query.php

```diff
@@ -203,4 +194,67 @@ private function resetPagination()
             ldap_set_option($con, \LDAP_OPT_SERVER_CONTROLS, $ctl);
         }
     }
+
+    protected function controlPagedResult($con, $pageSize, $isCritical = false, $cookie = '')
+    {
+        if (\PHP_VERSION_ID < 70300) {
+            return ldap_control_paged_result($con, $pageSize, $isCritical, $cookie);
+        }
+        $this->options['serverctrls'] = [
+            [
+                'oid' => \LDAP_CONTROL_PAGEDRESULTS,
+                'isCritical' => $isCritical,
+                'value' => [
+                    'size' => $pageSize,
+                    'cookie' => $cookie,
+                ],
+            ],
+        ];
+    }
+
+    protected function controlPagedResultResponse($con, $search, &$cookie = null)
```

Comment:
rather than creating an API using references, I suggest returning the cookie instead. That would better match the way we write code in Symfony.